File: client/ios/TemplatesSupportTests/Expected.swift
import Foundation

import CommonCorePublic

enum Expected {
  static let withRequiredProperty = EntityWithRequiredProperty(
    property: .value("Some text")
  )

  static let withOptionalProperty = EntityWithOptionalProperty(
    property: .value("Some text")
  )

  static let withMissingOptionalProperty = EntityWithOptionalProperty(
    property: nil
  )

  static let withDefaultValue = EntityWithPropertyWithDefaultValue(
    nested: EntityWithPropertyWithDefaultValue.Nested(
      nonOptional: .value("Some text")
    )
  )

  static let withRequiredComplexProperty = EntityWithComplexProperty(
    property: EntityWithComplexProperty.Property(
      value: .value(URL(string: "https://ya.ru")!)
    )
  )

  static let withOptionalComplexProperty = EntityWithOptionalComplexProperty(
    property: EntityWithOptionalComplexProperty.Property(
      value: .value(URL(string: "https://ya.ru")!)
    )
  )

  static let withMissingOptionalComplexProperty = EntityWithOptionalComplexProperty(
    property: nil
  )

  static let withArray = EntityWithArray(
    array: [
      .entityWithComplexProperty(withRequiredComplexProperty),
      .entityWithComplexProperty(withRequiredComplexProperty),
    ]
  )

  static let withStrictArray = EntityWithStrictArray(
    array: [
      .entityWithComplexProperty(withRequiredComplexProperty),
      .entityWithComplexProperty(withRequiredComplexProperty),
    ]
  )

  static let withHeterogeneousArray = EntityWithArray(
    array: [
      .entityWithComplexProperty(withRequiredComplexProperty),
      .entityWithRequiredProperty(withRequiredProperty),
      .entityWithStringEnumProperty(withStringEnumProperty),
    ]
  )

  static let withNestedArray = EntityWithArray(
    array: [
      .entityWithArray(withArray),
    ]
  )

  static let withArrayWithTransform = EntityWithArrayWithTransform(
    array: [
      .value(.colorWithHexCode(0x00_FF_00_FF)),
      .value(.colorWithHexCode(0xFF_00_00_AA)),
    ]
  )

  static let withStringEnumProperty = EntityWithStringEnumProperty(
    property: .value(.second)
  )

  static let withOptionalStringEnumProperty = EntityWithOptionalStringEnumProperty(
    property: .value(.second)
  )

  static let withMissingStringEnumProperty = EntityWithOptionalStringEnumProperty(
    property: nil
  )

  static let withArrayOfNestedItems = EntityWithArrayOfNestedItems(
    items: [
      EntityWithArrayOfNestedItems.Item(
        entity: .entityWithComplexProperty(withRequiredComplexProperty),
        property: .value("Some text")
      ),
      EntityWithArrayOfNestedItems.Item(
        entity: .entityWithComplexProperty(withRequiredComplexProperty),
        property: .value("Some text")
      ),
    ]
  )
}

File: client/android/api-generator-test/src/test/java/com/yandex/generator/Expected.kt
package com.yandex.generator

import android.graphics.Color
import android.net.Uri
import com.yandex.div.json.expressions.ConstantExpressionsList
import com.yandex.div.json.expressions.Expression
import com.yandex.testing.Entity
import com.yandex.testing.EntityWithArray
import com.yandex.testing.EntityWithArrayOfNestedItems
import com.yandex.testing.EntityWithArrayWithTransform
import com.yandex.testing.EntityWithComplexProperty
import com.yandex.testing.EntityWithOptionalComplexProperty
import com.yandex.testing.EntityWithOptionalProperty
import com.yandex.testing.EntityWithOptionalStringEnumProperty
import com.yandex.testing.EntityWithRequiredProperty
import com.yandex.testing.EntityWithStrictArray
import com.yandex.testing.EntityWithStringEnumProperty

internal val ENTITY_WITH_STRING_ENUM_PROPERTY = Entity.WithStringEnumProperty(
    EntityWithStringEnumProperty(Expression.constant(EntityWithStringEnumProperty.Property.SECOND))
)

internal val ENTITY_WITH_OPTIONAL_STRING_ENUM_PROPERTY = Entity.WithOptionalStringEnumProperty(
    EntityWithOptionalStringEnumProperty(Expression.constant(EntityWithOptionalStringEnumProperty.Property.SECOND))
)
internal val ENTITY_WITH_MISSING_STRING_ENUM_PROPERTY = Entity.WithOptionalStringEnumProperty(
    EntityWithOptionalStringEnumProperty(null)
)

internal val ENTITY_WITH_REQUIRED_PROPERTY = Entity.WithRequiredProperty(
    EntityWithRequiredProperty(Expression.constant("Some text"))
)

internal val ENTITY_WITH_OPTIONAL_PROPERTY = Entity.WithOptionalProperty(
    EntityWithOptionalProperty(Expression.constant("Some text"))
)

internal val ENTITY_WITH_MISSING_OPTIONAL_PROPERTY = Entity.WithOptionalProperty(
    EntityWithOptionalProperty(null)
)

internal val ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY = Entity.WithComplexProperty(
    EntityWithComplexProperty(EntityWithComplexProperty.Property(Expression.constant(Uri.parse("https://ya.ru"))))
)

internal val ENTITY_WITH_OPTIONAL_COMPLEX_PROPERTY = Entity.WithOptionalComplexProperty(
    EntityWithOptionalComplexProperty(EntityWithOptionalComplexProperty.Property(Expression.constant(Uri.parse("https://ya.ru"))))
)

internal val ENTITY_WITH_MISSING_OPTIONAL_COMPLEX_PROPERTY = Entity.WithOptionalComplexProperty(
    EntityWithOptionalComplexProperty(null)
)

internal val ENTITY_WITH_ARRAY = Entity.WithArray(
    EntityWithArray(listOf(
        ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY,
        ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY
    ))
)

internal val ENTITY_WITH_HETEROGENOUS_ARRAY = Entity.WithArray(
    EntityWithArray(listOf(
        ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY,
        ENTITY_WITH_REQUIRED_PROPERTY,
        ENTITY_WITH_STRING_ENUM_PROPERTY
    ))
)

internal val ENTITY_WITH_NESTED_ARRAY = Entity.WithArray(
    EntityWithArray(listOf(ENTITY_WITH_ARRAY))
)

internal val ENTITY_WITH_ARRAY_WITH_TRANSFORM = Entity.WithArrayWithTransform(
    EntityWithArrayWithTransform(ConstantExpressionsList(listOf(
        Color.parseColor("#FF00FF00"),
        Color.parseColor("#AAFF0000")
    )))
)

internal val ENTITY_WITH_ARRAY_OF_NESTED_ITEMS = Entity.WithArrayOfNestedItems(
    EntityWithArrayOfNestedItems(listOf(
        EntityWithArrayOfNestedItems.Item(ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY, Expression.constant("Some text")),
        EntityWithArrayOfNestedItems.Item(ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY, Expression.constant("Some text"))
    ))
)

internal val ENTITY_WITH_STRICT_ARRAY = Entity.WithStrictArray(
    EntityWithStrictArray(listOf(
        ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY,
        ENTITY_WITH_REQUIRED_COMPLEX_PROPERTY
    ))
)
